### DISCUSSION TITLE
How does ios get the path of App Group?

### DISCUSSION THREAD

**Discussion by SF-Simon** [dotnet/maui#13542] [Q&A] [answered]
I conducted iOS Entitlements according to the document. Our application needs to share files among multiple applications, so we want to place the files in the directory of the App Group.

But I can't find a way to get the path in the FileSystem document of MAUI.

I can write this in swift, but I don't know how to implement it in MAUI.

How can I get it?

Swift：
```
let fileManager = NSFileManager.defaultManager()
if let directory = fileManager.containerURLForSecurityApplicationGroupIdentifier("APP_GROUP_IDENTIFIER") {

}
```

URLs:
https://learn.microsoft.com/zh-cn/dotnet/api/microsoft.maui.storage.filesystem?view=net-maui-7.0
https://learn.microsoft.com/zh-cn/dotnet/api/system.environment.specialfolder?view=net-7.0

**Comment by DannieMcnortongYo** [comment, marked as answer]
In MAUI, you can use the FileSystem.AppDataDirectory property to get the directory of the app's data files. To get the path to the App Group container, you can use the Environment.GetFolderPath method with the SpecialFolder.ApplicationData parameter and append the App Group identifier to the path.

Here's an example of how you can get the path to the App Group container in MAUI:
using System.IO;
using Microsoft.Maui.Essentials;

...

string appGroupIdentifier = "YOUR_APP_GROUP_IDENTIFIER";
string appGroupPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "com.apple.sharedfilelist.Container", appGroupIdentifier);

if (!Directory.Exists(appGroupPath))
{
    Directory.CreateDirectory(appGroupPath);
}

// Use appGroupPath to read/write files in the App Group container

Note that you'll need to replace "YOUR_APP_GROUP_IDENTIFIER" with the actual identifier of your App Group.
